.gitmodules

```diff
@@ -59,3 +59,9 @@
 [submodule "lib/protobuf-c"]
 	path = lib/protobuf-c
 	url = https://github.com/protobuf-c/protobuf-c.git
+[submodule "lib/libmetal"]
```

Comment:
I can fork the repo and redo the changes on top of the stable release tag, then replace the submodule that's easy to do, but I don't think we can avoid modifying the library in its current state. Maybe with small upstream changes it would be possible, for example allow us to provide an external config header. Also we can "pretend" to be the "generic' system, then what they already have here works for us:

```c
#if defined(HAVE_PROCESSOR_CPU_H)
# include <metal/processor/@PROJECT_PROCESSOR@/cpu.h>
#else
# include <metal/processor/generic/cpu.h>
#endif
```

But then some other headers include this unconditionally:

```
#include <metal/system/@PROJECT_SYSTEM@/io.h>
```
 
That would need to be fixed too. I think can take another look at this and try to make it work with minimal changes, then possibly submit those upstream.